rest-api/src/main/java/gov/cms/qpp/conversion/api/controllers/v2/QrdaControllerV2.java

```diff
@@ -35,10 +38,16 @@ public QrdaControllerV2(QrdaService qrdaService, ValidationService validationSer
 	 * The version 2 API responds with validation warnings as well as the encoded QPP JsonWrapper.
 	 */
 	@Override
-	protected ConvertResponse respond(ConversionReport report) {
+	protected ConvertResponse respond(MultipartFile file, String checkedPurpose, HttpHeaders httpHeaders) {
+		ConversionReport conversionReport = buildReport(file.getOriginalFilename(), inputStream(file), checkedPurpose);
 		ConvertResponse response = new ConvertResponse();
-		response.setQpp(report.getEncoded().toObject());
-		response.setWarnings(report.getWarnings());
+		response.setQpp(conversionReport.getEncoded().toObject());
+		response.setWarnings(conversionReport.getWarnings());
+		Metadata metadata = audit(conversionReport);
+		if (null != metadata) {
+			httpHeaders.add("Location", metadata.getUuid());
+			response.setLocation(metadata.getUuid());
+		}
```

Comment:
Some of this code looks like copy paste from V1, could this common code be moved to the superclass?